cpu/cortexm_common/include/irq_arch.h

```diff
@@ -72,7 +73,7 @@ static inline __attribute__((always_inline)) int irq_is_enabled(void)
 /**
  * @brief See if the current context is inside an ISR
  */
-static inline __attribute__((always_inline)) int irq_is_in(void)
+static inline __attribute__((always_inline)) bool irq_is_in(void)
 {
     return (__get_IPSR() & 0xFF);
```

Comment:
this loses information if mapped to bool?

Reply:
It only ever was intended to return non-0 for `true` and 0 for `false`. 
There is nothing that does (or should) rely on the implementation specific value this returns.

Reply:
```
   71 /**                                                                                                                                          
   72  * @brief   Test if IRQs are currently enabled                                                                                               
   73  *                                                                                                                                           
   74  * @warning Use this function from thread context only. When used in interrupt                                                               
   75  *          context the returned state may be incorrect.                                                                                     
   76  *                                                                                                                                           
   77  * @return  0 (false) if IRQs are currently disabled                                                                                         
   78  * @return  != 0 (true) if IRQs are currently enabled                                                                                        
   79  */                                                                                                                                          
   80 MAYBE_INLINE int irq_is_enabled(void);                                                                                                       
```

It specifically says `!= 0` in the docs. :/

Reply:
`true` is `!= 0`
It does not specify which `!= 0` value to expect

Reply:
According to the C standard, `true` and `false` are macros that are 1 and 0. `bool` is also a macro and is simply an alias for `_Bool`. Furthermore, anything that is not 0 becomes 1 when assigned to `_Bool`. That is, whatever integer value is returned will become 1 if it is not 0. So if you assign any integer values that are not 0 to two boolean variables, the comparison will always be correct.

Maybe, the documentation should be changed to clarify that
```
 * @return  false (0) if IRQs are currently disabled                                                                                         
 * @return  true (1) if IRQs are currently enabled
```

Reply:
👍 but better use `@retval` for documenting individual return values.